src/lib.rs

```diff
@@ -333,18 +333,15 @@ where
 
                     let metrics_server = telemetry.serve_metrics(metrics_listener);
 
-                    let fut = ::logging::admin().bg("resolver").future(resolver_bg)
-                        .join5(
-                            ::logging::admin().bg("telemetry").future(telemetry),
-                            tap.map_err(|_| {}),
-                            metrics_server.map_err(|_| {}),
-                            ::logging::admin().bg("dns-resolver").future(dns_bg),
-                        )
-                        // There's no `Future::join6` combinator...
-                        .join(::logging::admin().bg("tls-config").future(tls_cfg_bg))
-                        .map(|_| {});
-
-                    rt.spawn(Box::new(fut));
+                    rt.spawn(::logging::admin().bg("resolver").future(resolver_bg));
+                    rt.spawn(::logging::admin().bg("telemetry").future(telemetry));
+                    // tap is already wrapped in a logging Future.
```

Comment:
Would it be problematic to make this consistent, so that either all of the wrapping in logging futures is done here, or it is all done when they are constructed? I'd slightly prefer the wrapping to be done here, but it seems like a good idea to at least be consistent or explain *why* we're not consistent.

Reply:
Peeking around, it seems those 2 (tap and metrics_server) are wrapping themselves after grabbing some relevant values from other arguments (such as the `local_addr`), and *then* making a clone of the logging context to be used for all internally spawned futures, so they share the context.

I just added these comments because *I* was also confused why it looked like some were wrapped and some were forgotten.